language/move-prover/src/boogie_wrapper.rs

```diff
@@ -818,29 +823,6 @@ impl<'env> BoogieWrapper<'env> {
             })
             .collect_vec()
     }
-
-    /// Extracts compilation errors. This captures any kind of errors different than the
```

Comment:
Looks like we would not longer display Boogie compilation errors using our own reporting mechanism. This was convenient for development. Did you check how this surfaces now?

Reply:
Indeed.

I deliberately introduced a bug in the Boogie code generator, dropping the ; after "assume !abort_flag".  I got the following output for arithm.move:

Error: boogie exited with: Output { status: ExitStatus(ExitStatus(0)), stdout: "output.bpl(1477,5): error: \";\" expected\noutput.bpl(1564,5): error: \";\" expected\noutput.bpl(1647,5): error: \";\" expected\noutput.bpl(1749,5): error: \";\" expected\noutput.bpl(1853,5): error: \";\" expected\noutput.bpl(2065,5): error: \";\" expected\noutput.bpl(2261,5): error: \";\" expected\noutput.bpl(2350,5): error: \";\" expected\noutput.bpl(2437,5): error: \";\" expected\noutput.bpl(2494,5): error: \";\" expected\noutput.bpl(2544,5): error: \";\" expected\noutput.bpl(2616,5): error: \";\" expected\noutput.bpl(2681,5): error: \";\" expected\noutput.bpl(2753,5): error: \";\" expected\noutput.bpl(2818,5): error: \";\" expected\noutput.bpl(2882,5): error: \";\" expected\noutput.bpl(2942,5): error: \";\" expected\noutput.bpl(3006,5): error: \";\" expected\noutput.bpl(3076,5): error: \";\" expected\noutput.bpl(3326,5): error: \";\" expected\noutput.bpl(3563,5): error: \";\" expected\noutput.bpl(3658,5): error: \";\" expected\noutput.bpl(3741,5): error: \";\" expected\noutput.bpl(3813,5): error: \";\" expected\noutput.bpl(3878,5): error: \";\" expected\noutput.bpl(3950,5): error: \";\" expected\noutput.bpl(4015,5): error: \";\" expected\noutput.bpl(4087,5): error: \";\" expected\noutput.bpl(4152,5): error: \";\" expected\noutput.bpl(4224,5): error: \";\" expected\noutput.bpl(4289,5): error: \";\" expected\noutput.bpl(4361,5): error: \";\" expected\noutput.bpl(4426,5): error: \";\" expected\noutput.bpl(4498,5): error: \";\" expected\noutput.bpl(4563,5): error: \";\" expected\noutput.bpl(4635,5): error: \";\" expected\noutput.bpl(4700,5): error: \";\" expected\noutput.bpl(4772,5): error: \";\" expected\noutput.bpl(4837,5): error: \";\" expected\noutput.bpl(4909,5): error: \";\" expected\noutput.bpl(4974,5): error: \";\" expected\noutput.bpl(5046,5): error: \";\" expected\noutput.bpl(5111,5): error: \";\" expected\noutput.bpl(5183,5): error: \";\" expected\noutput.bpl(5248,5): error: \";\" expected\noutput.bpl(5320,5): error: \";\" expected\noutput.bpl(5389,5): error: \";\" expected\noutput.bpl(5446,5): error: \";\" expected\n48 parse errors detected in output.bpl\n", stderr: "" }

It is not quite as pleasant as before but the necessary information for debugging is there.  I suppose that previously some line number information about the move source file was also printed which would be even more helpful.

Let me explain the reason I made this change.  Boogie dropped the BPxxxx code that was put in the error message.  This makes all errors look the same, which makes it difficult to distinguish between compilation and verification errors.  Instead of thinking about a more complicated regular expression, I took this brutal approach.

Any suggestions?

Reply:
Ok lets go ahead with this. And no, also previously no Move information was there for Boogie errors. As this is truly a breaking change once submitted, I think this needs to be well announced on move-prover and move-platform channels (everybody needs to run `./scripts/dev_setup.sh -yp` (https://github.com/diem/diem/blob/master/language/move-prover/doc/user/install.md)).